### PR TITLE
adding default department in case of bill

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/netsuite/settings/general-mappings/general-mappings.component.html [hunk 1]
```diff
@@ -47,6 +47,19 @@
              </mat-option>
            </mat-select>
          </mat-form-field>
+         <div class="mat-body-strong" *ngIf="generalSettings.corporate_credit_card_expenses_object === 'BILL'">
+          Netsuite Department
```
Comment (ashwin1111) [inline src/app/netsuite/settings/general-mappings/general-mappings.component.html:51]: NetSuite

File: src/app/netsuite/settings/general-mappings/general-mappings.component.html [hunk 2]
```diff
@@ -47,6 +47,19 @@
...
+        </div>
+        <mat-form-field floatLabel="always" appearance="outline" required
+          *ngIf="generalSettings.corporate_credit_card_expenses_object === 'BILL'">
+          <mat-select class="general-mappings--input" placeholder="Choose Netsuite Department"
```
Comment (ashwin1111) [inline src/app/netsuite/settings/general-mappings/general-mappings.component.html:55]: NetSuite

File: src/app/netsuite/settings/general-mappings/general-mappings.component.ts [hunk 3]
```diff
@@ -94,16 +99,23 @@ export class GeneralMappingsComponent implements OnInit {
     if (locationId === null) {
       that.locationIsValid = true;
     }
-
+    if (departmentId != null) {
```
Comment (ashwin1111) [inline src/app/netsuite/settings/general-mappings/general-mappings.component.ts:102]: !==

Comment (ashwin1111) [inline src/app/netsuite/settings/general-mappings/general-mappings.component.ts:102]: `if (departmentId) { ... }`

File: src/app/netsuite/settings/general-mappings/general-mappings.component.ts [hunk 4]
```diff
@@ -94,16 +99,23 @@ export class GeneralMappingsComponent implements OnInit {
...
+      that.departmentIsValid = true;
+    }
+    if (departmentId === null) {
```
Comment (ashwin1111) [inline src/app/netsuite/settings/general-mappings/general-mappings.component.ts:105]: `if (!departmentId) { ...}`

Comment (ashwin1111) [reply src/app/netsuite/settings/general-mappings/general-mappings.component.ts:105]: hey, this isn't correct. We are assigning `departmentIsValid` as true when departmentId is not null or even null. Check and remove it off. Add as required while creating form if it has to be validated. Can clean up old code as a separate task.

Comment (NileshPant1999) [reply src/app/netsuite/settings/general-mappings/general-mappings.component.ts:105]: yes even I was going to ask you and sravan about it.

**Pull request closed without merging**